experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,584 @@
+
+function Base.:(==)(x::CliffordAlgebraElem{T}, y::CliffordAlgebraElem{T}) where {T}
+  @req parent(x) === parent(y) "The inputs must lie in the same Clifford algebra"
+  return coefficients(x) == coefficients(y)
+end
+
+function Base.hash(x::CliffordAlgebraElem, h::UInt)
+  b = 0x1c4629b4de23b24c % UInt
+  h = hash(parent(x), h)
+  h = hash(coefficients(x), h)
+  return xor(h, b)
+end
+
+################################################################################
+#
+#  Graded parts
+#
+################################################################################
+
+@doc raw"""
+    even_part(x::CliffordAlgebraElem) -> CliffordAlgebraElem
+
+Return the projection of $x$ onto the even Clifford algebra
+"""
+even_part(x::CliffordAlgebraElem) = parent(x)(even_coefficients(x))
+
+@doc raw"""
+    odd_part(x::CliffordAlgebraElem) -> CliffordAlgebraElem
+
+Return the projection of $x$ onto the odd Clifford algebra.
+"""
+odd_part(x::CliffordAlgebraElem) = parent(x)(odd_coefficients(x))
+
+################################################################################
+#
+#  Auxillary functions
+#
+################################################################################
+
+function _set_even_odd_coefficients!(x::CliffordAlgebraElem)
+  x.even_coeffs = map(
+    y -> if sum(digits(y - 1; base=2, pad=_dim_qf(x.parent))) % 2 == 0
+      x.coeffs[y]
+    else
+      x.parent.base_ring()
+    end, 1:(x.parent.dim)
+  )
+  x.odd_coeffs = x.coeffs - x.even_coeffs
+  return x
+end
+
+function __can_convert_coefficients(coeff::Vector{R}, K::Field) where {R}
+  if length(coeff) == 0
+    return true
+  end
+  try
+    K.(coeff)  # Try converting the coefficient vector to K
+    true
+  catch
+    false
+  end
+end
+
+_dim_qf(C::CliffordAlgebra) = ncols(C.gram)
+
+function _mul_aux(x::Vector{T}, y::Vector{T}, gram::MatElem{T}, i::Int) where {T}
+  if length(y) == 1 #|| is_zero(y)
+    return x .* y[1]
+  end
+  return _mul_aux(_mul_with_gen(x, i, gram), y[2:2:end], gram, i + 1) +
+         _mul_aux(x, y[1:2:end], gram, i + 1)
+end
+
+#Implements the right multiplication with 'i'-th generator e_i of the Clifford algebra containing x.
+#The result is returned as a coefficient vector for further computations.
+_mul_with_gen(x::Vector{T}, i::Int, gram::MatElem{T}) where {T} = sum(
+  map(
+    j ->
+      x[j] .*
+      _mul_baseelt_with_gen(BitVector(digits(j - 1; base=2, pad=ncols(gram))), i, gram),
+    1:length(x),
+  ),
+)
+
+#In the following, bitvectors of length n are used, where 2^n is the Dimension of the Clifford algebra
+#containing x. A bitvector will represent the characteristic function of a subset I of {1,...,n}, thus
+#representing the element e_I of the canonical basis of the Clifford algebra. Here, e_I is the ordered
+#product of the elements of I.
+
+#Converts a bitvector to an integer. The first entry corresponds to 2^0, the second one to 2^1 and so on.
+function _bitvec_to_int(x::BitVector)
+  pow_of_two = 1
+  res = 0
```

Comment:
So you convert to `Int`... so at most 64 bits are allowed in your bit vector `x`.

So why do you use a `BitVector` at all?

Reply:
`BitVector` is not used anymore as of this [commit](https://github.com/oscar-system/Oscar.jl/pull/4370/commits/00b2cb1354677d596d83bfb70606be1b6eb3ce7b).